Mathlib/Algebra/Order/GroupWithZero/Range.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2025 Filippo A. E. Nuccio. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Filippo A. E. Nuccio
+-/
+import Mathlib.Algebra.GroupWithZero.Range
+import Mathlib.Algebra.Order.GroupWithZero.WithZero
+
+/-! # The range of a MonoidWithZeroHom
+
+Given a `MonoidWithZeroHom` `f : A → B` whose codomain `B` is a `LinearOrderedCommGroupWithZero`,
+we provide some order properties of the `MonoidWithZeroHom.valueGroup₀` as defined in
+`Mathlib.Algebra.GroupWithZero.Range`.
+
+-/
+
+namespace MonoidWithZeroHom
+
+variable {A B F : Type*} [FunLike F A B]
+variable [MonoidWithZero A] [LinearOrderedCommGroupWithZero B]
+variable [MonoidWithZeroHomClass F A B] {f : F}
```

Comment:
For definitions, we've been trying to avoid using hom classes, and instead preferring the actual hom. Lemmas should use the hom class, but data carrying decls shouldn't.

One reason is that you don't get any dot projection notation, and another is that solving for the class is harder, it always requires you to specify the types anyway. So I suggest making this `MonoidWithZeroHom.valueGroup_0_etc`

Reply:
OK, I tried but this does not seem to work very well. For instance, for the very first definition `valueMonoid` if I turn `f` to be a term of `MonoidWithZeroHom A B` in the definition and then I declare a variable `f : F` where `MonoidWithZeroHomClass F A B`, the coercion to `MonoidWithZerHom A B` does not fire automatically. If you want to check locally and suggest a compiling modification, I'd be happy to have a look.

Reply:
After multiple discussions on Zulip, I think I've got it, @YaelDillies...